lib/ansible/module_utils/urls.py

```diff
@@ -792,9 +792,18 @@ def maybe_add_ssl_handler(url, validate_certs):
                              ' however this is unsafe and not recommended')
 
         # do the cert validation
+        netloc = parsed.get('netloc')
+        if '@' in netloc:
+            netloc = netloc.split('@', 1)[1]
+        if ':' in netloc:
+            hostname, port = netloc.split(':', 1)
+            port = int(port)
+        else:
+            hostname = netloc
+            port = 443
         # create the SSL validation handler and
         # add it to the list of handlers
-        return SSLValidationHandler(parsed.get('hostname'), parsed.get('port'))
+        return SSLValidationHandler(hostname, port)
```

Comment:
When I use `return SSLValidationHandler(parsed.get('hostname'), parsed.get('port'))` the test is failing.